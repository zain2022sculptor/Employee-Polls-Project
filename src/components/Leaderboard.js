import { connect } from "react-redux";
import { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import {
  TableContainer,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Table,
} from "@mui/material";
import Login from "./Login";

const Leaderboard = (props) => {
  const navigate = useNavigate();

  useEffect(() => {
    if (props.authedUser === null) {
      //navigate("/");
    }
  }, []);

  return (
    <div>
      {props.authedUser === null ? (
        <Login />
      ) : (
        <div>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Avatars</TableCell>
                <TableCell>Users</TableCell>
                <TableCell>Answered</TableCell>
                <TableCell>Created</TableCell>
              </TableRow>
            </TableHead>
            {props.userIDs.map((userID) => {
              return (
                <TableBody key={userID}>
                  <TableRow>
                    <TableCell>
                      <img
                        src={props.users[userID].avatarURL}
                        alt={`Avatar of ${props.users[userID].name}`}
                        className="avatar center"
                      />
                    </TableCell>
                    <TableCell>
                      <div>{props.users[userID].name}</div>
                      <div>{userID}</div>
                    </TableCell>
                    <TableCell>
                      {Object.keys(props.users[userID].answers).length}
                    </TableCell>
                    <TableCell>
                      {props.users[userID].questions.length}
                    </TableCell>
                  </TableRow>
                </TableBody>
              );
            })}
          </Table>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = ({ users, authedUser }) => {
  const userIDs = Object.keys(users);
  userIDs.sort((a, b) => {
    const i = Object.keys(users[a].answers).length + users[a].questions.length;
    const j = Object.keys(users[b].answers).length + users[b].questions.length;
    return j - i;
  });

  return {
    userIDs,
    users,
    authedUser,
  };
};

export default connect(mapStateToProps)(Leaderboard);
