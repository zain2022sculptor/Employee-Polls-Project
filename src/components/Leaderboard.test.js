import { render } from "@testing-library/react";
import * as React from "react";
import { Provider } from "react-redux";
import Leaderboard from "./Leaderboard";
import { store } from "../index";
import { BrowserRouter } from "react-router-dom";

describe("Leaderboard", () => {
  it("will match the snapshot", () => {
    var view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Leaderboard />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
