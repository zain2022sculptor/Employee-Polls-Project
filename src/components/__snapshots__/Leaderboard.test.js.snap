// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard will match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <h1
            class="center"
          >
            Welcome to Employee Polls App
          </h1>
          <h3
            class="center"
          >
            Select user to Login
          </h3>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Select User
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="select-user"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Select User
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1hny3at-MuiButtonBase-root-MuiButton-root"
            data-testid="button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <h1
          class="center"
        >
          Welcome to Employee Polls App
        </h1>
        <h3
          class="center"
        >
          Select user to Login
        </h3>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
          >
            Select User
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            data-testid="select-user"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Select User
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1hny3at-MuiButtonBase-root-MuiButton-root"
          data-testid="button"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Login
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
